package/src/container/EdgeRenderer/EdgeRenderer.vue

```diff
@@ -79,12 +86,76 @@ const getType = (edge: GraphEdge) => {
 
   const slot = slots?.[`edge-${name}`]
   if (!slot?.({})) {
-    console.warn(`[vueflow]: Edge type "${edge.type}" not found and no edge-slot detected. Using fallback type "default".`)
+    console.warn(`[vueflow]: Edge type "${type}" not found and no edge-slot detected. Using fallback type "default".`)
     return false
   }
 
   return slot
 }
+
+const onEdgeClick = (event: MouseEvent, edge: GraphEdge) => {
+  const data = { event, edge }
+  if (selectable(edge.selectable)) {
+    setState({
+      nodesSelectionActive: false,
+    })
+
+    addSelectedEdges([edge])
+  }
+  emits.edgeClick(data)
+}
+
+const onEdgeContextMenu = (event: MouseEvent, edge: GraphEdge) => emits.edgeContextMenu({ event, edge })
+
+const onDoubleClick = (event: MouseEvent, edge: GraphEdge) => emits.edgeDoubleClick({ event, edge })
+
+const onEdgeMouseEnter = (event: MouseEvent, edge: GraphEdge) => emits.edgeMouseEnter({ event, edge })
+
+const onEdgeMouseMove = (event: MouseEvent, edge: GraphEdge) => emits.edgeMouseMove({ event, edge })
+
+const onEdgeMouseLeave = (event: MouseEvent, edge: GraphEdge) => emits.edgeMouseLeave({ event, edge })
+
+const onEdgeUpdaterSourceMouseDown = (event: MouseEvent, edge: GraphEdge) => handleEdgeUpdater(event, edge, true)
+
+const onEdgeUpdaterTargetMouseDown = (event: MouseEvent, edge: GraphEdge) => handleEdgeUpdater(event, edge, false)
+
+const { onMouseDown } = useHandle()
+
+const handleEdgeUpdater = (event: MouseEvent, edge: GraphEdge, isSourceHandle: boolean) => {
+  const nodeId = isSourceHandle ? edge.target : edge.source
+  const handleId = (isSourceHandle ? edge.targetHandle : edge.sourceHandle) ?? ''
+
+  emits.edgeUpdateStart({ event, edge })
+
+  onMouseDown(
+    event,
+    handleId,
+    nodeId,
+    isSourceHandle,
+    undefined,
+    isSourceHandle ? 'target' : 'source',
+    (connection) => {
+      if (!connectionExists(edge, getEdges)) emits.edgeUpdate({ edge, connection })
```

Comment:
Should be ` if (!connectionExists(connection, getEdges)) emits.edgeUpdate({ edge, connection }) `